test/specs/deployAndRetrieve.e2e.ts

```diff
@@ -5,26 +5,331 @@
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Deploy This Source to Org', 5);
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(
+      workbench,
+      'Running SFDX: Deploy Source to Org',
+      utilities.FIVE_MINUTES
+    );
+    const successNotificationWasFound = await utilities.notificationIsPresent(
+      workbench,
+      'SFDX: Deploy Source to Org successfully ran'
+    );
+    expect(successNotificationWasFound).toBe(true);
+
+    // Verify Output tab
+    const outputPanelText = await utilities.attemptToFindOutputPanelText(
+      'Salesforce CLI',
+      'Starting SFDX: Deploy Source to Org',
+      10
+    );
+    utilities.log('Deploy time - 2: ' + (await utilities.getOperationTime(outputPanelText!)));
+    expect(outputPanelText).not.toBeUndefined();
+    expect(outputPanelText).toContain('Deployed Source');
+    expect(outputPanelText).toContain(`Unchanged  MyClass    ApexClass  ${pathToClass}.cls`);
+    expect(outputPanelText).toContain(
+      `Unchanged  MyClass    ApexClass  ${pathToClass}.cls-meta.xml`
+    );
+    expect(outputPanelText).toContain('ended SFDX: Deploy Source to Org');
+  });
+
+  step('Modify the file and deploy again', async () => {
+    // Clear the Output view first.
+    const outputView = await utilities.openOutputView();
+    await outputView.clearText();
+
+    // Modify the file by adding a comment.
+    const workbench = await browser.getWorkbench();
+    const editorView = await workbench.getEditorView();
+    const textEditor = (await editorView.openEditor('MyClass.cls')) as TextEditor;
+    await textEditor.setTextAtLine(2, '\t//say hello to a given name');
+    await textEditor.save();
+
+    // Deploy running SFDX: Deploy This Source to Org
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Deploy This Source to Org', 5);
+
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(
+      workbench,
+      'Running SFDX: Deploy Source to Org',
+      utilities.FIVE_MINUTES
+    );
+    const successNotificationWasFound = await utilities.notificationIsPresent(
+      workbench,
+      'SFDX: Deploy Source to Org successfully ran'
+    );
+    expect(successNotificationWasFound).toBe(true);
+
+    // Verify Output tab
+    const outputPanelText = await utilities.attemptToFindOutputPanelText(
+      'Salesforce CLI',
+      'Starting SFDX: Deploy Source to Org',
+      10
+    );
+    utilities.log('Deploy time - 3: ' + (await utilities.getOperationTime(outputPanelText!)));
+    expect(outputPanelText).not.toBeUndefined();
+    expect(outputPanelText).toContain('Deployed Source');
+    expect(outputPanelText).toContain(`Changed  MyClass    ApexClass  ${pathToClass}.cls`);
+    expect(outputPanelText).toContain(`Changed  MyClass    ApexClass  ${pathToClass}.cls-meta.xml`);
+    expect(outputPanelText).toContain('ended SFDX: Deploy Source to Org');
```

Comment:
To compare the deploy duration, How about we repeat the above deploy steps with the source tracking setting disabled?

Reply:
And we need to turn on source tracking in the org setting.